### DISCUSSION TITLE
Python error when model is being trained

### DISCUSSION THREAD

**Discussion by VisionedWheat66** [mindsdb/mindsdb#4958] [Get help]
I have installed mindsdb v23.3.2.0 in a Proxmox LXC container running Debian 11. After creating the container I have installed pip, created a virtual environment and installed mindsdb in it letting it add all dependencies it wants.

I connected via the http API and successfully connected to my data source. I then created a model but when it begins training it fails with this error recorded in the models table:

AttributeError: module 'A2JDXXBL9A1E1678749699474425' has no attribute 'Predictor', raised at: /root/mindsdb-venv/lib/python3.9/site-packages/mindsdb/integrations/libs/ml_exec_base.py#131

The full stack trace of this error is attached as taken from the console: [stack trace.txt](https://github.com/mindsdb/mindsdb/files/10962695/stack.trace.txt)

Given the "out of box" nature of this setup (I don't think I've missed anything out above in the steps I've done), does anyone have any idea on what could be going wrong here? I've searched on here and wider on Google but there are no relevant results for the phrase "no attribute 'Predictor'" so I'm not sure why I'm getting this error when clearly it works for everyone else.

Thanks for any help.

**Comment by chandrevdw31** [comment]
@VisionedWheat66 can you kindly share the `CREATE MODEL` query you tried to execute please.
Feel free to join our MindsDB Slack community https://mindsdb.com/joincommunity . You will be able to engage with our community members and MindsDB Team. Thank you.

**Comment by ZoranPandovski** [comment]
Hey @VisionedWheat66 which API you have used to train the model? Are you using some of our SDKs?

**Comment by VisionedWheat66** [comment]
Hi guys, thanks for the replies. Thanks for the community link @chandrevdw31, I'll check that out.

@ZoranPandovski I'm connecting to a MariaDB database set up via the http frontend. Not using any SDKs for anything, all that's present is what gets pulled by pip when installing.

Here's the create model statement:
```sql
CREATE MODEL mindsdb.SolarGeneration
FROM EnergyUsage
    (SELECT *
     FROM EnergyUsage.VW_SolarGeneratedPredictor)
PREDICT SolarGenerated
```

The view `EnergyUsage.VW_SolarGeneratedPredictor` is in a MariaDB database and returns 61169 rows with these column types:

```
Date | date
WeatherMoment | datetime
EnergyUsageMoment | datetime
SolarGenerated | decimal(6,3)
SunriseUTC | datetime
SunsetUTC | datetime
WeatherConditionCode | tinyint(3) UN
```

When I execute the create model statement it runs for about as long as it takes the view to finish returning results, then I get the stack trace from my original post in the console.

I'll keep investigating when possible and appreciate the help guys, thank you.
